bgpd/bgpd.c

```diff
@@ -1176,6 +1177,8 @@ static void peer_free(struct peer *peer)
 
 	memset(peer, 0, sizeof(struct peer));
 
+	if (peer->ibuf_scratch)
+		XFREE(MTYPE_BGP_IBUF_SCRATCH, peer->ibuf_scratch);
```

Comment:
We never check a pointer value before freeing it, `XFREE` does this already

Reply:
ok